clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreAwarePartitionWiseKafkaConsumerService.java

```diff
@@ -0,0 +1,196 @@
+   * for different consumer. For batch only store, resource identifier is topic name. For hybrid store, resource
+   * identifier is the store name.
+   */
+  private final Map<String, Map<PubSubConsumerAdapter, Integer>> resourceIdentifierToConsumerMap =
+      new VeniceConcurrentHashMap<>();
+  private final ReadOnlyStoreRepository storeRepository;
+
+  StoreAwarePartitionWiseKafkaConsumerService(
+      final ConsumerPoolType poolType,
+      final PubSubConsumerAdapterFactory consumerFactory,
+      final Properties consumerProperties,
+      final long readCycleDelayMs,
+      final int numOfConsumersPerKafkaCluster,
+      final IngestionThrottler ingestionThrottler,
+      final KafkaClusterBasedRecordThrottler kafkaClusterBasedRecordThrottler,
+      final MetricsRepository metricsRepository,
+      final String kafkaClusterAlias,
+      final long sharedConsumerNonExistingTopicCleanupDelayMS,
+      final TopicExistenceChecker topicExistenceChecker,
+      final boolean liveConfigBasedKafkaThrottlingEnabled,
+      final PubSubMessageDeserializer pubSubDeserializer,
+      final Time time,
+      final AggKafkaConsumerServiceStats stats,
+      final boolean isKafkaConsumerOffsetCollectionEnabled,
+      final ReadOnlyStoreRepository metadataRepository,
+      final boolean isUnregisterMetricForDeletedStoreEnabled) {
+    super(
+        poolType,
+        consumerFactory,
+        consumerProperties,
+        readCycleDelayMs,
+        numOfConsumersPerKafkaCluster,
+        ingestionThrottler,
+        kafkaClusterBasedRecordThrottler,
+        metricsRepository,
+        kafkaClusterAlias,
+        sharedConsumerNonExistingTopicCleanupDelayMS,
+        topicExistenceChecker,
+        liveConfigBasedKafkaThrottlingEnabled,
+        pubSubDeserializer,
+        time,
+        stats,
+        isKafkaConsumerOffsetCollectionEnabled,
+        metadataRepository,
+        isUnregisterMetricForDeletedStoreEnabled,
+        StoreAwarePartitionWiseKafkaConsumerService.class.toString());
+    this.storeRepository = metadataRepository;
+  }
+
+  @Override
+  protected synchronized SharedKafkaConsumer pickConsumerForPartition(
+      PubSubTopic versionTopic,
+      PubSubTopicPartition topicPartition) {
+    String resourceIdentifier = getResourceIdentifier(versionTopic);
+    int minLoad = Integer.MAX_VALUE;
+    SharedKafkaConsumer minLoadConsumer = null;
+    Map<Integer, Integer> consumerToLoadMap = new VeniceConcurrentHashMap<>();
+    for (SharedKafkaConsumer consumer: getConsumerToConsumptionTask().keySet()) {
+      int index = getConsumerToConsumptionTask().indexOf(consumer);
+      if (topicPartition.getPubSubTopic().isRealTime()
+          && alreadySubscribedRealtimeTopicPartition(consumer, topicPartition)) {
+        getLOGGER().info(
+            "Consumer id: {} has already subscribed the same real time topic-partition: {}, will assign MAX load to avoid being picked.",
```

Comment:
Nit,
```
will assign MAX load to avoid being picked
```
I guess we should mention this consumer will be `skipped`, right?

Reply:
Yes, I can update the doc to be more clear.